cmd/rule-evaluator/main.go

```diff
@@ -67,6 +67,16 @@ import (
 
 const projectIDVar = "PROJECT_ID"
 
+var googleCloudBaseURL = urlMustParse("https://console.cloud.google.com/monitoring/metrics-explorer")
```

Comment:
suggestion: store the url string as a const `string`.

Then we can use it as the default value for `query.generator-url` in the flag declaration along with the `projectID` defaulting like we do [elsewhere](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/0c597767c1883093107b0f875fca14bafd9480de/cmd/rule-evaluator/main.go#L111).

We can then build/parse the full URL struct [later](https://github.com/GoogleCloudPlatform/prometheus-engine/blob/0c597767c1883093107b0f875fca14bafd9480de/cmd/rule-evaluator/main.go#L153-L158) to be passed to `sendAlerts` as-is without changing its function signature.

Reply:
I'm not sure the `projectID` defaulting makes sense here because the `projectID` field is not enough. Should we encode the expression in the URL? How about the start and end times? More importantly, where does what gets encoded into the URL end? How do we document this API?

I personally feel like this would be a bit of a feature-creep to add when you consider all the possibilities. Fortunately, how this PR was designed, we always have the possibility to add it in, in a backwards-compatible fashion.

Reply:
That's a very good point. I still feel strongly against adding templating, so I went with a different approach: the default is still the raw URL but we add the project manually if we detect it.